### PR TITLE
Add benchmark and Test for scan of Unfold

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes : #1313

File: test/Streamly/Test/Data/Unfold.hs [hunk 1]
```diff
@@ -449,6 +450,17 @@ enumerateFromToFractional =
 -- Stream transformation
 -------------------------------------------------------------------------------
 
+scan :: Property
+scan =
+    property
+        $ \f _list ->
+              let _fA = apply f :: Int -> Int
+                  _x = _list :: [Int]
```
Comment (harendra-kumar) [inline test/Streamly/Test/Data/Unfold.hs:458]: This looks ugly, why is `_list` even being passed as argument? Why is `_fA` required.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: test/Streamly/Test/Data/Unfold.hs [hunk 2]
```diff
@@ -449,6 +450,17 @@ enumerateFromToFractional =
...
+                  unf = UF.scan Fold.sum UF.fromList
+                  list = [1,2,3,4] :: [Int]
+                  mList = [1,3,6,10] :: [Int]
```
Comment (harendra-kumar) [inline test/Streamly/Test/Data/Unfold.hs:461]: Can we use the passed list instead and compare it against the Data.List scan function?

File: test/Streamly/Test/Data/Unfold.hs [hunk 3]
```diff
@@ -449,6 +450,14 @@ enumerateFromToFractional =
 -- Stream transformation
 -------------------------------------------------------------------------------
 
+scan :: Property
+scan =
+    property
+        $ \(ls :: [Int]) ->
+              let unf = UF.scan Fold.sum UF.fromList
+                  mList = scanl1 (+) ls
```
Comment (harendra-kumar) [inline test/Streamly/Test/Data/Unfold.hs:458]: Looks like `UF.scan` should be renamed to `UF.postscan`.

Also, I noticed that `UF.scanlM'` is marked as `Unimplemented` even though it is implemented. We can fix that too. Also. UF.scanlM' should be renamed as `postscanlM'`.

Also, rename this test as `postscan`.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Pull request merged**